### ISSUE TITLE
Export collisions not considered errors when in .d.ts files

### ISSUE THREAD

**Issue by thesoftwarephilosopher** [microsoft/TypeScript#61553]
### 🔎 Search Terms

export collision error

### 🕗 Version & Regression Information

- This changed between versions ______ and _______
- This changed in commit or PR _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because _______

### 💻 Code

a.ts

```ts
export const foo = 3
```

b.ts

```ts
export const foo = 3
```

c.d.ts

```ts
export * from './a.js'
export * from './b.js'
```

### 🙁 Actual behavior

No error raised by tsc.

### 🙂 Expected behavior

Consider this an error per https://github.com/microsoft/TypeScript/issues/4527

### Additional information about the issue

Prior discussion: https://stackoverflow.com/questions/79562798/get-typescript-to-warn-on-export-conflicts

**Comment by RyanCavanaugh** [comment]
Turn off skipLibCheck

**Comment by thesoftwarephilosopher** [comment]
Oops.

**Comment by thesoftwarephilosopher** [comment]
Wait doesn't lib check include everything in node_modules? If so, I might incur a heavy performance penalty just because I want *my* .d.ts files checked...

**Comment by RyanCavanaugh** [comment]
Right, this is a trade-off. FWIW you should almost never be hand-authoring `.d.ts` files; I can't think of a situation where you'd (correctly) be writing `c.d.ts`

**Comment by thesoftwarephilosopher** [comment]
I write very unusual stuff. In this case, it's auto-generated by a build tool as a placeholder for a .js file that's also auto-generated by it.

**Comment by thesoftwarephilosopher** [comment]
> Right, this is a trade-off. FWIW you should almost never be hand-authoring `.d.ts` files; I can't think of a situation where you'd (correctly) be writing `c.d.ts`

One I've had several times is when I have a foo.js file that will exist at runtime but not dev time, so if I make foo.d.ts then I can import it and use the expected exports in it. I expect this will be more common as bundling moves towards vendoring thanks to ESM.
